plugin.video.wizjatv/main.py

```diff
@@ -33,55 +36,38 @@ def WizjaTV():
     s, content = obj.ListaKanalow()
     content = json.loads(content)
     for item in content:
-        addDir(item['title'],item['url'] , 6, item['icon'],'','', False)
+        addDir(name=item['title'], url=item['url'], mode=PLAY_LINK, thumb=item['icon'], isFolder=False)
 
-def OdpalanieLinku():
+def OdpalanieLinku(url):
     try:
         global s
-        url = urllib.unquote_plus(params['url'])
         s, content = obj.ListaKanalow()
-        link = obj.Link(url,s)
+        link = obj.Link(url, s)
         xbmc.Player().play(str(link).replace("rtmp://$OPT:rtmp-raw=", ""))
     except:
         log_exception()
-        pass
 
 def get_params():
     paramstring = sys.argv[2]
     if paramstring.startswith('?'):
         paramstring = paramstring[1:]
-    return dict((k, vv[0]) for k, vv in parse_qs(paramstring).items())
+    return dict((k, urllib.unquote_plus(vv[0])) for k, vv in parse_qs(paramstring).items())
 
 params = get_params()
-url = None
-name = None
-thumb = None
-mode = None
-iconimage = None
+url = params.get('url')
+name = params.get('name')
+iconimage = params.get('iconimage')
 
-try:
-    url = urllib.unquote_plus(params['url'])
-except:
-    pass
-try:
-    name = params.get('name')
-except:
-    pass
 try:
     mode = int(params.get('mode'))
-except:
+except (TypeError, ValueError):
     mode = None
     log_exception()
-    pass
-try:
-    iconimage = urllib.unquote_plus(params['iconimage'])
-except:
-    pass
 
-if mode == None :
+if mode is None:
     CATEGORIES()
-elif mode == 6 : 
-    OdpalanieLinku()
+elif mode == PLAY_LINK:
```

Comment:
Konsekwencja, czyli brak magic numer i poniżej przekazanie do funkcji uzyskanego już url.

Reply:
Chyba warto we wszystkich wtyczkach wywalić magic number i zastąpić go nazwą określającą zachowanie. Kod będzie czytelniejszy. Jak myślisz?

Reply:
Tak, jak najbardziej. To jeden z kroków „ucywilizowania” naszego kodu :)

Reply:
W sumie to nie ma żadnego wymogu (znaczy ja go nie widzę), żeby `mode` musiało być liczbą. Możemy we wtyczkach wywalić liczby i użyć wprost nazwy. W tym przypadku wyciąć `PLAY_LINK = 6` używać np.

```python
addDir(name=item['title'], url=item['url'], mode='play', thumb=item['icon'], isFolder=False)
...
elif mode == 'play': 
      OdpalanieLinku(url)
```

Nazwa dowolna, byle by określała co to mniej więcej robi.